blueprints/auto-unseal/main.tf

```diff
@@ -0,0 +1,343 @@
+    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+    token                  = data.aws_eks_cluster_auth.this.token
+  }
+}
+
+data "aws_eks_cluster_auth" "this" {
+  name = module.eks_blueprints.eks_cluster_id
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_availability_zones" "available" {}
+
+locals {
+  cluster_name    = var.name
+  cluster_version = var.cluster_version
+  
+  vpc_name        = var.name
+  vpc_cidr        = var.vpc_cidr
+  azs             = slice(data.aws_availability_zones.available.names, 0, 2)
+
+  tags = {
+    Blueprint  = "blueprints/${var.name}"
+    GithubRepo = "github.com/hashicorp/terraform-aws-hashicorp-vault-eks-addon"
+  }
+
+}
+
+################################################################################
+# EKS Blueprints Cluster
+################################################################################
+
+module "eks_blueprints" {
+  # See https://github.com/aws-ia/terraform-aws-eks-blueprints/releases for latest version
+  # Example is not pinned to avoid update cycle conflicts between module and implementation
+  # tflint-ignore: terraform_module_pinned_source
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints"
+
+  # EKS CONTROL PLANE VARIABLES
+  cluster_name    = local.cluster_name
+  cluster_version = local.cluster_version
+
+  # EKS Cluster VPC and Subnet mandatory config
+  vpc_id             = module.vpc.vpc_id
+  private_subnet_ids = module.vpc.private_subnets
+
+  # EKS MANAGED NODE GROUPS
+  managed_node_groups = {
+    mg = {
+      node_group_name = "managed-spot"
+      capacity_type   = "SPOT"
+      instance_types  = ["t3.small", "t3a.small", "t3.medium","t3.large"]
+      min_size        = 1
+      max_size        = 5
+      desired_size    = 3
+      subnet_ids      = module.vpc.private_subnets
+    }
+  }
+
+  tags = local.tags
+}
+
+################################################################################
+# EKS Blueprints Addons
+################################################################################
+
+module "eks_blueprints_kubernetes_addons" {
+  # See https://github.com/aws-ia/terraform-aws-eks-blueprints/releases for latest version
+  # Example is not pinned to avoid update cycle conflicts between module and implementation
+  # tflint-ignore: terraform_module_pinned_source
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons"
+
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
+
+  # EKS Managed Add-ons
+  enable_amazon_eks_vpc_cni           = true
+  enable_amazon_eks_coredns           = true
+  enable_amazon_eks_kube_proxy        = true
+  enable_aws_load_balancer_controller = true
+  enable_karpenter                    = true
+
+  # HashiCorp Vault
+  enable_vault = true
+  vault_helm_config = {
+    name       = "vault"
+    chart      = "vault"
+    repository = "https://helm.releases.hashicorp.com"
+    version    = "v0.23.0"
```

Comment:
Recommend bumping the default version in the module to `v0.23.0` and removing the defaults for name, chart, repository.

Reply:
Updated.